sourcegraph/sourcegraph
# Toggle visibility of line series

Closes: https://github.com/sourcegraph/sourcegraph/issues/29958

[Designs](https://www.figma.com/file/fWW7rO9nUVZJEqUZXTMxcX/Click-to-turn-on%2Foff-a-line-series-%2329958-%5Bready-for-dev%5D?node-id=1617%3A2217)

https://user-images.githubusercontent.com/1855233/169619393-cf6303cd-b4e9-48ef-99e1-0731cd7c44b0.mov

## Test plan

- Existing tests should pass
- Existing screenshots should render the same
- Open any line chart with multiple series. Easiest would be to open storybook and go to the "Smart Insights View Grid Example"
- Click a legend item
- Only that series should be rendered.
- Other legend items should be faded
- Hover over another legend item, and its respective series should render slightly faded

## App preview:

- [Web](https://sg-web-insights-click-series-toggle.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-zeucsshbcf.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
